### PR TITLE
Init frontend

### PR DESCRIPTION
**Pull request by rexolion**
## Description
Init React/Typescript Project for Union + eslint.

## Related issue(s)
None

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

- Public contracts
  - [x] My changes doesn't break any Licenses.
  - [ ] I added an entry to the [changelog](../tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [ ] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Union](https://github.com/union-platform/guides/blob/master/commits.md).
- [x] My code complies with the [styles guide](https://github.com/union-platform/guides/blob/master/README.md#programming-languages).

#### Before merging (mandatory)
- [ ] Check for `fixup!` or `squash!` commits in this branch
- [ ] Check __target__  branch of PR is set correctly

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -4,3 +4,26 @@
 
 node_modules/
 dist/
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
```
Comment (worm2fed) [inline frontend/.gitignore:10]: this is a file, isn't it?
and in this file upper we've already have `node_modules/`

File: frontend/.gitignore [hunk 2]
```diff
@@ -4,3 +4,26 @@
...
+/.pnp
+.pnp.js
```
Comment (worm2fed) [inline frontend/.gitignore:12]:
```
/.pnp
.pnp.js
```

why not `.pnp*`?

File: frontend/.gitignore [hunk 3]
```diff
@@ -4,3 +4,26 @@
...
+
+# testing
+/coverage
```
Comment (worm2fed) [inline frontend/.gitignore:15]: if it's a directory, it should be `coverage/`

File: frontend/.gitignore [hunk 4]
```diff
@@ -4,3 +4,26 @@
...
+
+# production
+/build
```
Comment (worm2fed) [inline frontend/.gitignore:18]: same

File: frontend/.gitignore [hunk 5]
```diff
@@ -4,3 +4,26 @@
...
+
+# misc
+.DS_Store
```
Comment (worm2fed) [inline frontend/.gitignore:21]: please, do not add your OS specific files, to ignore it - add to your global OS gitignore

File: frontend/.gitignore [hunk 6]
```diff
@@ -4,3 +4,26 @@
...
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
```
Comment (worm2fed) [inline frontend/.gitignore:25]: `.env.*.local`?

File: frontend/.gitignore [hunk 7]
```diff
@@ -4,3 +4,26 @@
...
+
+npm-debug.log*
```
Comment (worm2fed) [inline frontend/.gitignore:27]: `*.log*`

File: frontend/package.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "union_frontend",
```
Comment (worm2fed) [inline frontend/package.json:2]: let's name it `union-app`? what do you think?

File: frontend/package.json [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  "version": "0.1.0",
```
Comment (worm2fed) [inline frontend/package.json:3]: Is haskell there are practice to set `0.0.0` until package is released and then, for first release set `0.1.0` (or smth)
Is it same for frontend staff?

File: frontend/public/index.html [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (worm2fed) [inline frontend/public/index.html:27]: Please, set real title here, for example `Union` or `Union Platform`

File: frontend/public/index.html [hunk 11]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (worm2fed) [inline frontend/public/index.html:10]: Please, set real description here

File: frontend/public/robots.txt [hunk 12]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (worm2fed) [inline frontend/public/robots.txt:3]: do we want to disallow robots parse Union? I guess it's cool to show our app in search engines

**Review by worm2fed — CHANGES_REQUESTED** [review]
Also - please add license identifiers to all your files

**Comment by worm2fed** [comment]
and for your future PRs
Please, set branch name as described in our policy
instead of `init_frontend` it should be `rexolion/init_frontend`, and in case it would have issue associated with your pr, it should be `rexolion/#420-init_frontend`

File: frontend/public/robots.txt (hunk 12 above)
Comment (rexolion) [reply frontend/public/robots.txt:3]: Actually "Disallow: " (empty) means "Allow all", disallow all — "Disallow: /" — https://www.robotstxt.org/robotstxt.html

File: frontend/.reuse/dep5 [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# SPDX-FileCopyrightText: 2021 Union
+# SPDX-License: AGPL-3.0-or-later
+
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+
+Files: *
```
Comment (worm2fed) [inline frontend/.reuse/dep5:6]: Since you've added licence identifiers to frontend files, I think you do not need this `/frontend/.reuse/` dir any more)

**Review by worm2fed — APPROVED** [review]
It's good now.
But please, squash your commits and add appropreate body to first one, like here: https://github.com/union-platform/guides/blob/master/commits.md#commit-messages (point 4)

**Review by worm2fed — APPROVED** [review]

**Pull request merged**